src/main/resources/META-INF/resources/primefaces/datepicker/1-datepicker.js

```diff
@@ -318,6 +318,7 @@ PrimeFaces.widget.DatePicker = PrimeFaces.widget.BaseWidget.extend({
             var viewChangeBehavior = this.cfg.behaviors['viewChange'];
 
             if(viewChangeBehavior) {
+                options.update = (options.update||'') +' '+ this.id;
```

Comment:
Not sure if you need to add `process` too?

Reply:
I'm not sure. Did some minor improvements and tested with the showcase PR I submitted (I've added events) and it's all working for me.